src/omega_edit/omega_edit.js

```diff
@@ -88,6 +85,32 @@ function saveSession() {
   })
 }
 
+function replace(data, offset, len) {
+  vscode.postMessage({
+    command: 'replace',
+    data: data,
+    offset: offset,
+    len: len,
+  })
+}
+
+function searchSession(searchPattern, caseInsensitive) {
+  vscode.postMessage({
+    command: 'search',
+    searchPattern: searchPattern,
+    caseInsensitive: caseInsensitive,
+  })
+}
+
+function searchAndReplace(searchPattern, replaceText, caseInsensitive) {
+  vscode.postMessage({
+    command: 'searchAndReplace',
+    searchPattern: searchPattern,
+    replaceText: replaceText,
+    caseInsensitive: caseInsensitive,
+  })
+}
+
 function setupViewport(viewportId) {
   viewportElement = document.getElementById(viewportId)
   viewportElement.addEventListener('keydown', (event) => {
```

Comment:
Hmmm. Keydown event for specific keys, feels like a user-configurable JSON file should specify the key bindings used, not straight code. 

Fine if that's in a future plan, but I'm going to want to swtich all these bindings to be EMACS-like.

Reply:
So I think with the webview you can't make a json with key mappings. From what I was able to find this is the only may I could get the key bindings to work using a webview.